library/ssl_tls13_client.c

```diff
@@ -1364,9 +1371,18 @@ static int ssl_tls13_is_supported_versions_ext_present(
     MBEDTLS_SSL_CHK_BUF_READ_PTR(p, end, legacy_session_id_echo_len + 4);
     p += legacy_session_id_echo_len + 4;
 
-    return mbedtls_ssl_tls13_is_supported_versions_ext_present_in_exts(
-        ssl, p, end,
-        &supported_versions_data, &supported_versions_data_end);
+    if ((ret = mbedtls_ssl_tls13_is_supported_versions_ext_present_in_exts(
+             ssl, p, end,
+             &supported_versions_data, &supported_versions_data_end)) != 1) {
+        return ret;
+    }
+    ret = ssl_tls13_parse_supported_versions_ext(ssl,
```

Comment:
Why do you want to do this earlier? Regarding version negotiation as far as I remember just checking the presence of the extension is enough.

Reply:
Yes.  I think that is enough also.

I move it here because of `Clients MUST check for this extension prior to processing the rest of the ServerHello` from `RFC 8446 section 4.2.1`. Should we follow-up it ? I am not very sure.

Reply:
I just remove it and add downgrade protection magic words change.